batch/plugins/org.jboss.tools.batch.ui/src/org/jboss/tools/batch/ui/quickfixes/BatchQuickFixGenerator.java

```diff
@@ -0,0 +1,262 @@
+			int end = start + tempAnnotation.getPosition().getLength();
+			
+			try {
+				return findXMLResolutions(file, messageId, start, end, true);
+			} catch (JavaModelException e) {
+				BatchUIPlugin.getDefault().logError(e);
+			}
+		}
+		return new IJavaCompletionProposal[]{};
+	}
+
+	@Override
+	public boolean hasResolutions(IMarker marker) {
+		if(marker.exists()){
+			try {
+				return getMessageID(marker) >= 0;
+			} catch (CoreException ex) {
+				BatchUIPlugin.getDefault().logError(ex);
+			}
+		}
+		return false;
+	}
+	
+	private int getMessageID(TemporaryAnnotation annotation){
+		if(annotation.getAttributes() != null){
+			Integer attribute = ((Integer) annotation.getAttributes().get(ValidationErrorManager.MESSAGE_ID_ATTRIBUTE_NAME));
+			if (attribute != null)
+				return attribute.intValue();
+		}
+		return -1; 
+	}
+	
+	private int getMessageID(IMarker marker)throws CoreException{
+		Integer attribute = ((Integer) marker.getAttribute(ValidationErrorManager.MESSAGE_ID_ATTRIBUTE_NAME));
+		if (attribute != null)
+			return attribute.intValue();
+
+		return -1; 
+	}
+	
+	private IMarkerResolution[] findResolutions(IMarker marker)
+			throws CoreException {
+
+		int messageId = getMessageID(marker);
+		if (messageId == -1)
+			return new IMarkerResolution[] {};
+
+		if(marker.getResource() instanceof IFile){
+			final IFile file = (IFile) marker.getResource();
+	
+			Integer attribute = ((Integer) marker.getAttribute(IMarker.CHAR_START));
+			if (attribute == null)
+				return new IMarkerResolution[] {};
+			final int start = attribute.intValue();
+			
+			attribute = ((Integer) marker.getAttribute(IMarker.CHAR_END));
+			if (attribute == null)
+				return new IMarkerResolution[] {};
+			final int end = attribute.intValue();
+			
+			if (XML_EXTENSION.equals(file.getFileExtension())){
+				return findXMLResolutions(file, messageId, start, end, false);
+			}
+		}
+		return new IMarkerResolution[] {};
+	}
+	
+	private IJavaElement findJavaElementByQualifiedName(IJavaProject javaProject, String qualifiedName){
+		try {
+			return javaProject.findType(qualifiedName);
+		} catch (JavaModelException ex) {
+			BatchUIPlugin.getDefault().logError(ex);
+		}
+		
+		return null;
+	}
+	
+	private IQuickFix[] findXMLResolutions(IFile file, int messageId, int start, int end, boolean asYouType) throws JavaModelException{
+		IJavaProject javaProject = EclipseUtil.getJavaProject(file.getProject());
+		
+		FileEditorInput input = new FileEditorInput(file);
+		IDocumentProvider provider = DocumentProviderRegistry.getDefault().getDocumentProvider(input);
+		try {
+			provider.connect(input);
+		} catch (CoreException e) {
+			BatchUIPlugin.getDefault().logError(e);
+		}
+		
+		IDocument document = provider.getDocument(input);
+		
+		String text="";
+		try {
+			text = document.get(start, end-start);
+		} catch (BadLocationException e) {
+			BatchUIPlugin.getDefault().logError(e);
+		} finally {
+			provider.disconnect(input);
```

Comment:
provider.connect(input); and provider.disconnect(input); should be in the same try{} finally{} block.